pillar/fluentd/server_operations_qa.sls

```diff
@@ -48,40 +40,42 @@ fluentd:
             - '@id': heroku_logs_inbound
             - '@type': heroku_syslog_http
             - tag: heroku_logs
-            - bind: ::1
             - port: 9000
-            - protocol_type: tcp
-            - nested_directives:
-                - directive: parse
-                  attrs:
-                    - message_format: rfc5424
+            - bind: ::1
         - directive: source
           attrs:
             - '@id': mailgun-events
             - '@type': http
             - port: 9001
             - bind: ::1
-            - format: json
         - directive: source
           attrs:
             - '@id': salt_logs_inbound
             - '@type': udp
             - tag: saltmaster
             - format: json
             - port: 9999
-            - keep_time_key: 'true'
+            - bind: ::1
+            - nested_directives:
+                - directive: parse
+                  attrs:
+                    - keep_time_key: 'true'
         - directive: source
           attrs:
-            - '@id': secure_input
             - '@type': forward
             - port: 5001
+            - bind: ::1
             - nested_directives:
-              - directive: transport
-                directive_arg: tls
-                attrs:
-                  - cert_path: '/etc/ssl/certs/log-input.crt'
-                  - private_key_path: '/etc/ssl/certs/log-input.key'
-                  - private_key_passphrase: ''
+                - directive: security
+                  attrs:
+                    - self_hostname: {{ salt.grains.get('external_ip') }}
+                - directive: transport
+                  directive_arg: tls
+                  attrs:
+                    - cert_path: /etc/fluent/fluentd.crt
+                    - private_key_path: /etc/fluent/fluentd.key
```

Comment:
If we're going to put the keys/certs under /etc/fluent then we should probably have a state to ensure that ownership and access to that directory is only available to the user running the fluent process.

Reply:
Yeah was planning on doing that bit through the fluentd formula that will be creating those files.